src/Compilers/CSharp/Portable/Binder/ExecutableCodeBinder.cs

```diff
@@ -107,14 +107,21 @@ public static void ValidateIteratorMethod(CSharpCompilation compilation, MethodS
                 return;
             }
 
-            foreach (var parameter in iterator.Parameters)
+            var parameters = !iterator.IsStatic
+                ? iterator.GetParametersIncludingExtensionParameter()
```

Comment:
Should the `GetParametersIncludingExtensionParameter` helper check that the member is not static?

Reply:
That would not be convenient. That method is also used in overload resolution, and for the purpose of overload resolution, we do want to account for the extension parameter even when the method is static (the static receiver counts as an argument in that case).